src/gui/userstatus.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) by Camila <hello@camila.codes>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef USERSTATUS_H
+#define USERSTATUS_H
+
+#include <QObject>
+#include <QPointer>
+#include <QVariant>
+
+namespace OCC {
+class AccountState;
+class JsonApiJob;
+
+class UserStatus : public QObject
+{
+    Q_OBJECT
+public:
+    explicit UserStatus(AccountState *accountState, QObject *parent = nullptr);
+    QString currentStatus() const;
+    void fetchCurrentStatus();
+
+private slots:
+    void slotFetchedCurrentStatus(const QJsonDocument &json);
+    void slotRequestFailed();
+
+signals:
+    void fetchedCurrentStatus(UserStatus *userStatus);
+
+private:
+    QPointer<AccountState> _accountState;
```

Comment:
I personally don't like using QPointer except for some use cases. Here I don't see why it would be needed given that _accountState always outlives UserStatus the way it's currently used

Reply:
I guess I don't mind `QPointer` as much as @nicolasfella. That said I agree it feels unneeded in that context.